### PR TITLE
Don't use DataGrid for empty states (Closes #1636)

### LINKED ISSUE #1636: Datagrid can't handle empty semesters
When creating a new semester its semester overview page in staff mode loads and throws a JavaScript error: `this.sortableHeaders.get(...) is undefined`.

The datagrid should be fixed to handle an empty semester correctly.

### PR DESCRIPTION
**Pull request by Skn0tt**
Another way to solve #1636. Supercedes #1642.

### CODE REVIEW

File: evap/staff/templates/staff_semester_view.html [hunk 1]
```diff
@@ -576,18 +584,24 @@ <h3>
...
-            storageKey: "course-semester-{{ semester.id }}-data-grid",
-            table: document.querySelector("#course-table"),
-            searchInput: document.querySelector("input[name=search-course]"),
-            resetSearch: document.querySelector("[data-reset=search-course]"),
-        }).init();
+        const evaluationTable = document.querySelector("#evaluation-table");
+        if (evaluationTable) {
+            new EvaluationGrid({
+                storageKey: "evaluation-semester-{{ semester.id }}-data-grid",
+                table: document.querySelector("#evaluation-table"),
```
Comment (niklasmohrin) [inline evap/staff/templates/staff_semester_view.html:591]:
```suggestion
                table: evaluationTable,
```

same with `courseTable` below

File: evap/staff/templates/staff_semester_view.html [hunk 2]
```diff
@@ -262,17 +262,21 @@ <h3>
...
+                                <tr id="evaluation-row-{{ evaluation.id }}">
+                                    {% include 'staff_semester_view_evaluation.html' with semester=semester evaluation=evaluation info_only=False %}
+                                </tr>
+                            {% endfor %}
+                            </tbody>
+                        </table>
+                    {% else %}
+                        <table>
+                            <tbody>
+                                <tr><td><i>{% trans 'There are no evaluations in this semester.' %}</i><td></tr>
```
Comment (janno42) [inline evap/staff/templates/staff_semester_view.html:276]:
```suggestion
                                <tr><td><i>{% trans 'There are no evaluations in this semester.' %}</i></td></tr>
```
not your fault, but as you're on it anyway...

File: evap/staff/templates/staff_semester_view.html [hunk 3]
```diff
@@ -350,51 +354,55 @@ <h3>
...
+                                        {% endif %}
+                                    </td>
+                                </tr>
+                            {% endfor %}
+                        </tbody>
+                    </table>
+                {% else %}
+                    <table class="table table-striped table-narrow table-vertically-aligned">
+                        <tbody>
                             <tr><td><i>{% trans 'There are no courses in this semester.' %}</i><td></tr>
```
Comment (janno42) [inline evap/staff/templates/staff_semester_view.html:402]:
```suggestion
                            <tr><td><i>{% trans 'There are no courses in this semester.' %}</i></td></tr>
```

File: evap/staff/templates/staff_semester_view.html [hunk 4]
```diff
@@ -262,17 +262,21 @@ <h3>
...
+                            <tbody>
+                            {% for evaluation in evaluations %}
+                                <tr id="evaluation-row-{{ evaluation.id }}">
+                                    {% include 'staff_semester_view_evaluation.html' with semester=semester evaluation=evaluation info_only=False %}
+                                </tr>
+                            {% endfor %}
+                            </tbody>
+                        </table>
+                    {% else %}
+                        <table>
```
Comment (janno42) [inline evap/staff/templates/staff_semester_view.html:274]:
```suggestion
                        <table class="table table-striped table-narrow table-vertically-aligned">
```
the styling should be kept

File: evap/staff/templates/staff_semester_view.html [hunk 5]
```diff
@@ -576,18 +584,24 @@ <h3>
...
+                searchInput: document.querySelector("input[name=search-evaluation]"),
+                filterButtons: [...document.querySelectorAll("#evaluation-filter-buttons [data-filter]")],
+                resetSearch: document.querySelector("[data-reset=search-evaluation]"),
+            }).init();
+        }
+        const courseTable = document.querySelector("#course-table");
+        if (courseTable) {
+            new TableGrid({
+                storageKey: "course-semester-{{ semester.id }}-data-grid",
+                table: document.querySelector("#course-table"),
```
Comment (janno42) [inline evap/staff/templates/staff_semester_view.html:601]:
```suggestion
                table: course-table,
```

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

**Review by He3lixxx — APPROVED** [review]

**Pull request merged**